lib/src/protocol/http/parser/mod.rs

```diff
@@ -263,11 +263,12 @@ fn status_line(i: &[u8]) -> IResult<&[u8], StatusLine> {
 }
 
 #[derive(PartialEq, Debug)]
-pub struct Header<'a> {
-    pub name: &'a [u8],
-    pub value: &'a [u8],
+pub struct Header {
+    pub name: Vec<u8>,
```

Comment:
this will result in a massive performance regression. The parser is designed to avoid allocations as much as possible, while this will introduce two allocations per header, even for headers that are not used by the HTTP implementation.